src/test/java/com/jnape/palatable/lambdakoans/AboutApplicatives.java

```diff
@@ -0,0 +1,157 @@
+package com.jnape.palatable.lambdakoans;
+
+import com.jnape.palatable.lambda.adt.Either;
+import com.jnape.palatable.lambda.adt.Maybe;
+import com.jnape.palatable.lambda.adt.hlist.Tuple2;
+import com.jnape.palatable.lambda.functions.Fn1;
+import com.jnape.palatable.lambda.functions.Fn3;
+import com.jnape.palatable.lambda.functions.builtin.fn3.LiftA2;
+import com.jnape.palatable.lambda.functions.builtin.fn4.LiftA3;
+import com.jnape.palatable.lambda.functor.Applicative;
+import com.jnape.palatable.lambda.functor.builtin.Lazy;
+import com.jnape.palatable.lambda.io.IO;
+import com.jnape.palatable.lambda.traversable.LambdaIterable;
+import org.junit.Test;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static com.jnape.palatable.lambda.adt.Either.left;
+import static com.jnape.palatable.lambda.adt.Either.right;
+import static com.jnape.palatable.lambda.adt.Maybe.just;
+import static com.jnape.palatable.lambda.adt.Maybe.nothing;
+import static com.jnape.palatable.lambda.adt.hlist.HList.tuple;
+import static com.jnape.palatable.lambda.functions.Fn1.fn1;
+import static com.jnape.palatable.lambda.functions.builtin.fn1.Repeat.repeat;
+import static com.jnape.palatable.lambda.functions.builtin.fn2.Take.take;
+import static com.jnape.palatable.lambda.functor.builtin.Lazy.lazy;
+import static com.jnape.palatable.lambdakoans.Koans.__;
+import static java.util.Arrays.asList;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static testsupport.matchers.IterableMatcher.iterates;
+
+public class AboutApplicatives {
+
+    @Test
+    public void applicativesZipEmbeddedValuesWithEmbeddedFunctions() {
+        Fn1<Integer, Integer> inc = x -> x + 1;
+
+        assertThat(just(9).zip(just(inc)), equalTo(just(10)));
+        assertThat(just(15).zip(just(inc)), equalTo(just(__)));
+        // Explicit types help the compiler
+        assertThat(Maybe.<Integer>nothing().zip(just(inc)), equalTo(__));
+
+        Fn1<? super String, ? extends Integer> getStringLength = String::length;
+        assertThat(right("Hello").zip(right(getStringLength)), equalTo(__));
+        assertThat(right("World!").zip(right(getStringLength)), equalTo(__));
+        assertThat(Either.<String, String>left("whoops").zip(right(getStringLength)), equalTo(left(__)));
+
+        // Moving on to LambdaIterables, where "zipping" is more obvious
```

Comment:
We might want to point out that `LambdaIterable` is a specific choice in terms of how to implement Zip for iterables. It could be pairwise or cross-product. Alternatively, it might be worth putting it in a separate "test" to make that explicit.

Reply:
Added a comment about that possible alternative in [5d1d6da](https://github.com/palatable/lambda-koans/pull/6/commits/5d1d6da6ea86833c9c96bd5b07835c541cfae51a).